packages/ownable/src/lib.rs

```diff
@@ -89,73 +269,37 @@ pub fn initialize_owner(
     api: &dyn Api,
     owner: Option<&str>,
 ) -> StdResult<Ownership<Addr>> {
-    let ownership = Ownership {
-        owner: owner.map(|h| api.addr_validate(h)).transpose()?,
-        pending_owner: None,
-        pending_expiry: None,
-    };
-    OWNERSHIP.save(storage, &ownership)?;
-    Ok(ownership)
+    OWNERSHIP.initialize_owner(storage, api, owner)
 }
 
 /// Return Ok(true) if the contract has an owner and it's the given address.
 /// Return Ok(false) if the contract doesn't have an owner, of if it does but
 /// it's not the given address.
 /// Return Err if fails to load ownership info from storage.
 pub fn is_owner(store: &dyn Storage, addr: &Addr) -> StdResult<bool> {
-    let ownership = OWNERSHIP.load(store)?;
-
-    if let Some(owner) = ownership.owner {
-        if *addr == owner {
-            return Ok(true);
-        }
-    }
-
-    Ok(false)
+    OWNERSHIP.is_owner(store, addr)
 }
 
 /// Assert that an account is the contract's current owner.
 pub fn assert_owner(store: &dyn Storage, sender: &Addr) -> Result<(), OwnershipError> {
-    let ownership = OWNERSHIP.load(store)?;
-    check_owner(&ownership, sender)
-}
-
-/// Assert that an account is the contract's current owner.
-fn check_owner(ownership: &Ownership<Addr>, sender: &Addr) -> Result<(), OwnershipError> {
-    // the contract must have an owner
-    let Some(current_owner) = &ownership.owner else {
-        return Err(OwnershipError::NoOwner);
-    };
-
-    // the sender must be the current owner
-    if sender != current_owner {
-        return Err(OwnershipError::NotOwner);
-    }
-
-    Ok(())
+    OWNERSHIP.assert_owner(store, sender)
 }
 
 /// Update the contract's ownership info based on the given action.
 /// Return the updated ownership.
 pub fn update_ownership(
-    deps: DepsMut,
+    api: &dyn Api,
```

Comment:
This is a breaking change, we should keep it as `deps` to not break the API.

Sadly I published the `2.0` versions before seeing this PR, otherwise it would have made sense.

Reply:
dang. what about retagging  and republishing `2.0`?

Reply:
Unfortunate but in the interest of "just making it work" I reverted this and made the matching updates to `cw-nfts` in my PR there.